### PR TITLE
don't clone types that are copy

### PR DESCRIPTION
**Pull request by matthiaskrgr**
Found via clippy.

r? @eddyb

### CODE REVIEW

File: src/librustc_ast_lowering/expr.rs [hunk 1]
```diff
@@ -909,18 +909,18 @@ impl<'hir> LoweringContext<'_, 'hir> {
 
     fn lower_expr_asm(&mut self, asm: &InlineAsm) -> hir::ExprKind<'hir> {
         let inner = hir::InlineAsmInner {
-            inputs: asm.inputs.iter().map(|&(ref c, _)| c.clone()).collect(),
+            inputs: asm.inputs.iter().map(|&(ref c, _)| *c).collect(),
```
Comment (eddyb) [inline src/librustc_ast_lowering/expr.rs]: Can the closure be `|&(c, _)| c`?

File: src/libtest/lib.rs [hunk 2]
```diff
@@ -485,9 +485,7 @@ pub fn run_test(
         }
         StaticBenchFn(benchfn) => {
             // Benchmarks aren't expected to panic, so we run them all in-process.
-            crate::bench::benchmark(desc, monitor_ch, opts.nocapture, |harness| {
-                (benchfn.clone())(harness)
-            });
+            crate::bench::benchmark(desc, monitor_ch, opts.nocapture, |harness| (benchfn)(harness));
```
Comment (eddyb) [inline src/libtest/lib.rs]: You don't need the parens around `benchfn` - in fact, you might be able to just pass `benchfn` to `crate::bench::benchmark` directly, without the closure wrapping.

**Review by eddyb — APPROVED** [review]
r=me with nits fixed

**Comment by matthiaskrgr** [comment]
Addressed comments.

File: src/librustc_ast_lowering/expr.rs (hunk 1 above)
Comment (matthiaskrgr) [reply src/librustc_ast_lowering/expr.rs]: Yeah, that works.

File: src/libtest/lib.rs (hunk 2 above)
Comment (matthiaskrgr) [reply src/libtest/lib.rs]: Updated.

**Pull request merged**